### DISCUSSION TITLE
RAD and NTSC C64

### DISCUSSION THREAD

**Discussion by flyboyn222pa** [frntc/RAD#31] [Q&A]
I'm having trouble getting the RAD to boot on multiple NTSC C64's with a Raspberry Pi 3A+.  I've extracted the firmware zip file onto the root directory of multiple SD cards, and when I connect the HDMI output of the Pi I get no video signal.  If I power the RAD and then boot the C64. it simply boots to BASIC and I don't get a RAD menu.  The reset on the RAD works, but the menu button doesn't do anything.  I've tried changing the rad.cfg file to get it to boot directly into REU mode, but that didn't do anything either.  Most people on here talk about changing settings to get various screens to work, but I can't even get the RAD menu to boot.  I haven't seen any signs of life from my RAD/Raspberry Pi 3A+ combination.  I have tested the Pi with Raspberry Pi OS and that works normally.

Does anyone have any idea where I am going wrong?

**Comment by frntc** [comment]
When you boot the RAD and turn the C64 on -- does the green LED on the RPi fade in and out? It reads like the firmware is not booting (it's normal that HDMI doesn't output anything, also that "reset" works, because it's like any hardware reset button)

**Comment by flyboyn222pa** [comment]
It doesn’t fade in and out, the green LED flashes 4 times slowly and then 7 times quickly before repeating. On Oct 13, 2023, at 11:18 AM, frntc ***@***.***> wrote:﻿
When you boot the RAD and turn the C64 on -- does the green LED on the RPi fade in and out? It reads like the firmware is not booting (it's normal that HDMI doesn't output anything, also that "reset" works, because it's like any hardware reset button)

**Reply by khalim745** [reply]
Hello,

I have same failure with Raspberry Pi 3A+, 4 long flashes then 7 short flashes on the green LED instead of boot.
Then I tried with a Raspberry Pi 3B+, it was working inmediatelly.
The Raspberry Pi OS also working on this 3A+. Now I can send back the replacement pcb, because the 1st one
was also good.

**Reply by frntc** [reply]
That's weird. I can't imagine why the firmware should boot on a 3B+ but not on a 3A+ (I most often use the latter). And even less I could imagine how this would related to NTSC :-)

**Comment by khalim745** [comment]
Hello,

The pi 3B+ was brand new, out from the box,
the pi 3A+ was also new, but tried before with installation of Raspberry Pi
OS. Maybe during this installation something was updated in the firmware of
3A+.My C64 is not NTSC it’s a C64C Pal version.But this 3A+ RAD setup was
tested in a C128 also
with same results.

Best wishes,

khalim745

frntc ***@***.***> (időpont: 2023. okt. 27., P, 15:52) ezt
írta:

**Reply by frntc** [reply]
ok, sounds as if something is wrong with the 3A+. I've never done anything with Pi OS, so I can't tell if it does some "harm".

**Comment by khalim745** [comment]
Maybe the default GPU memory allocation is too high, there’s no enough
memory for run start.elf, there’s only 512MB on A+ this is the main
difference.

frntc ***@***.***> (időpont: 2023. okt. 27., P, 16:20) ezt
írta:

**Reply by frntc** [reply]
In config.txt there is gpu_mem=16, so this should not be the problem.

**Comment by khalim745** [comment]
Hello,

The raspi-config tool detects it as raspberry pi 3A+ rev.1.1 and there’s no
option in advanced settings to change gpu memory split, this point is
completely missing.

All the best

khalim745


frntc ***@***.***> (időpont: 2023. okt. 28., Szo, 15:30) ezt
írta:

**Reply by varmfskii** [reply]
I have has the same problem with two different new (2022 revision) Pi 3A+. They work fine for Raspberry Pi OS. If I swap for an older revision Pi 3A+ the issue does not exist.

The 2022 revision uses a different power control chip and start.elf is supposed to take into consideration this different chip.

**Reply by frntc** [reply]
This came also to my mind. In this case, it should work again once the "Raspberry Pi Firmware" on the SD-card is updated. It can be found here: 

https://github.com/raspberrypi/firmware/tree/e2bab29767e51c683a312df20014e3277275b8a6/boot

(take the files which you find in the root directory of the SD-card and replace them by the ones with the same filename from above - make sure you download the raw-data, e.g. using "wget -q -O bootcode.bin .../bootcode.bin?raw=true" )

Please let me know if this solves the problems you experience.

**Reply by varmfskii** [reply]
It solves the issue on both the RAD and the Sidekick64. Thank You.

**Reply by frntc** [reply]
great, thanks!

**Reply by SensoryTrash** [reply]
I am having the same problem with the 2022 pi3a+  any chance of getting a link to the files as i didnt have any luck making my own.... or does the Firmware need to be replaced on the PI itself

**Reply by Gamesmonkey** [reply]
Hi 
I also have the same problem with the Pi 3 A+
Iam not well versed with how you setup a Pi and don't really understand what you have said above.

Could you elaborate please.

Thanks

P.s I have put together a SideKick64  with Pi 3 B+ which works great. 😀

**Comment by varmfskii** [comment]
I'm on the road. I'll be home soon and will get y'all the appropriate files.

Get Outlook for Android<https://aka.ms/AAb9ysg>
________________________________
From: Gamesmonkey ***@***.***>
Sent: Tuesday, January 2, 2024 4:49:31 PM
To: frntc/RAD ***@***.***>
Cc: Evans, Alex ***@***.***>; Comment ***@***.***>
Subject: [External] Re: [frntc/RAD] RAD and NTSC C64 (Discussion #31)


Hi
I also have the same problem with the Pi 3 A+
Iam not well versed with how you setup a Pi and don't really understand what you have said above.

Could you elaborate please.

Thanks

P.s I have put together a SideKick64 with Pi 3 B+ which works great. 😀

**Comment by SensoryTrash** [comment]
you could try this file...

[RAD_Firmware_v011Pi 2022.zip](https://github.com/frntc/RAD/files/13820086/RAD_Firmware_v011Pi.2022.zip)

**Reply by Gamesmonkey** [reply]
Hi,

Thank you this solved the problem and now boots up.
